e2e-tests/src/main/java/uk/nhs/adaptors/gp2gp/Mongo.java

```diff
@@ -24,8 +28,10 @@ private static MongoDatabase prepareDatabaseConnection() {
         if (sharedDatabaseConnection == null) {
             var connectionString = System.getenv().getOrDefault("GP2GP_MONGO_URI", "mongodb://localhost:27017");
             var database = System.getenv().getOrDefault("GP2GP_MONGO_DATABASE_NAME", "gp2gp");
-            var client = create(connectionString);
-            sharedDatabaseConnection = client.getDatabase(database);
+
+            try (var client = create(connectionString)) {
```

Comment:
Will this dispose of the client after line 34? Is that what we want?

Reply:
yeah, you are right, it is too early to dispose the client in this place, good call, I will revert the changes.